### PR TITLE
Add clientStatus to Presence Model

### LINKED ISSUE #76: Investigate "online on mobile" feature
Discord added a new mobile indicator to the client. The backend sends a modified `PRESENCE_UPDATE` event (from what I've heard there's a new field).

Investigate that and implement it into Yasmin (requires modification of `Presence` model).

### PR DESCRIPTION
**Pull request by arcdigital**
**Pull Request description:**
Add clientStatus to the Presence model. clientStatus is an associative array that can have any combination of `web`, `mobile`, and `desktop` each with their own status.

**Why should this Pull Request be merged:**
Closes #76

### CODE REVIEW

**Comment by CharlotteDunois** [comment]
Thanks for filing this PR! 

Would you mind creating a new model class called `ClientStatus` instead of just taking whatever Discord sends? This might look like a neat little shortcut on the front, but taking into account of previous experience with Discord and their inconsistency and eventual consistency, there is no guarantee that each element in the array is present.
Presenting a class however to the user gives at least the guarantees that if the `clientStatus` property is not null, each property at least exists even if they can be null and as such removes the need for additional checks.

**Comment by arcdigital** [comment]
Sure, I'll try and do that tomorrow.

**Comment by arcdigital** [comment]
@CharlotteDunois can you take a look now?

File: src/Models/ClientStatus.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * Website: https://charuru.moe
+ * License: https://github.com/CharlotteDunois/Yasmin/blob/master/LICENSE
+*/
+
+namespace CharlotteDunois\Yasmin\Models;
+
+/**
+ * Discord Client Status.
+ *
+ * @property string|null        $desktop            The status of the user on the desktop client.
```
Comment (CharlotteDunois) [inline src/Models/ClientStatus.php]: You dont need that many spaces here :)

File: src/Models/ClientStatus.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * @property string|null        $mobile             The status of the user on the mobile client.
+ * @property string|null        $web                The status of the user on the web client.
+ */
+class ClientStatus extends Base {
+    /**
+     * The status of the user on the desktop client.
+     */
+    protected $desktop;
```
Comment (CharlotteDunois) [inline src/Models/ClientStatus.php]: Missing `@var`

File: src/Models/ClientStatus.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    protected $mobile;
+    
+    /**
+     * The status of the user on the web client.
+     * @var string|null
+     */
+    protected $web;
+
+    /**
+     * @param array             $clientStatus      An array containing the client status data.
```
Comment (CharlotteDunois) [inline src/Models/ClientStatus.php]: Missing method description (`Constructor.` is fine`. And also no need for that many spaces, you can minimize it to 2 spaces on each side.

File: src/Models/ClientStatus.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        
+        return parent::__get($name);
+    }
+
+    /**
+     * @return mixed
+     * @internal
+     */
+    function jsonSerialize() {
```
Comment (CharlotteDunois) [inline src/Models/ClientStatus.php]: Either remove this method or implement the necessary interface `JsonSerializable`. I'm fine with either way.

Comment (arcdigital) [reply src/Models/ClientStatus.php]: I'm extending the `Base` model, which already implements `JsonSerializable`.

**Comment by arcdigital** [comment]
@CharlotteDunois I think I addressed all your comments.

File: src/Models/ClientStatus.php (hunk 3 above)
Comment (arcdigital) [reply src/Models/ClientStatus.php]: Fixed.

File: src/Models/ClientStatus.php (hunk 2 above)
Comment (arcdigital) [reply src/Models/ClientStatus.php]: Fixed.

File: src/Models/ClientStatus.php (hunk 1 above)
Comment (arcdigital) [reply src/Models/ClientStatus.php]: Fixed.

**Comment by CharlotteDunois** [comment]
Thank you!

**Pull request merged**